scss/components/dialog.scss

```diff
@@ -0,0 +1,121 @@
+@import "./../settings";
+@import "./../mixins";
+@import "./../functions";
+@import "./../icons/mixins";
+
+/*!
+.fd-dialog
+    .fd-dialog__header
+        .fd-form__title
+    .fd-form__body
+    .fd-form__footer-items
+      .fd-dialog__button-primary
+      .fd-dialog__button-secondary
+*/
+
+
+$block: #{$fd-namespace}-dialog;
+  $fd-dialog-width--small: 288px !default;
+  $fd-dialog-width--medium: 460px !default;
+  $fd-dialog-width--phone: 320px !default;
+  $fd-dialog-header-height: 48px !default;
+  $fd-dialog-footer-height: 48px !default;
+
+  $fd-dialog-border-color: fd-color("neutral", 2) !default;
+  $fd-dialog-padding-x: var(--fd-width-gutter) !default;
+  $fd-dialog-padding-y: fd-space(3) !default;
+  $fd-dialog-inner-content-background: fd-color("background", 2) !default;
+
+  
+.#{$block}{
+  @include fd-reset;
+  
+  &__content{
+    border-radius: $fd-border-radius;
+    background-color: $fd-dialog-inner-content-background;
+  }
+
+  &__header {
+    position: relative;
+    border-bottom: 1px solid $fd-dialog-border-color;
+    padding: $fd-dialog-padding-y $fd-dialog-padding-x;
+  }
+  
+  &__header-error{
+    @extend .#{$block}__header;
```

Comment:
These extends are not necessary since the markup should include the base class. See above. The modifier should have two dashes `--`.